staging/src/k8s.io/cloud-provider/controllers/node/node_controller.go

```diff
@@ -295,39 +416,19 @@ func (cnc *CloudNodeController) updateNodeAddress(ctx context.Context, node *v1.
 // in a retry-if-conflict loop.
 type nodeModifier func(*v1.Node)
 
-func (cnc *CloudNodeController) UpdateCloudNode(ctx context.Context, _, newObj interface{}) {
-	node, ok := newObj.(*v1.Node)
-	if !ok {
-		utilruntime.HandleError(fmt.Errorf("unexpected object type: %v", newObj))
-		return
-	}
-
-	cloudTaint := getCloudTaint(node.Spec.Taints)
-	if cloudTaint == nil {
-		// The node has already been initialized so nothing to do.
-		return
-	}
-
-	cnc.initializeNode(ctx, node)
-}
-
-// AddCloudNode handles initializing new nodes registered with the cloud taint.
-func (cnc *CloudNodeController) AddCloudNode(ctx context.Context, obj interface{}) {
-	node := obj.(*v1.Node)
+// SyncNode handles updating existing nodes registered with the cloud taint
+// and processes nodes that were added into the cluster, and cloud initialize them if appropriate.
+func (cnc *CloudNodeController) SyncNode(ctx context.Context, node *v1.Node) {
```

Comment:
Since on line 454 we're fetching the latest node from cache, I think syncNode should be updated to receive a node name and not the entire node object, and likewise we can remove getting a node from cache on line 239